vpr/src/route/rr_graph2.cpp

```diff
@@ -1132,8 +1098,14 @@ static void load_block_rr_indices(RRGraphBuilder& rr_graph_builder,
                 int root_x = x - width_offset;
```

Comment:
Could the whole double-for loop comtaining these calls be moved into RRGraphBuilder as a high-level API?  Call it something like dupLargeBlkLookups().  Then mirror_nodes could be removed or made a (private if possible) helper function.  

Right now the comment for mirror_node describes a good chunk of the body of this for loop, despite the for loop being in the caller.  I think that's a sign we really want a higher-level API.

Reply:
Yes. This is what I thought:

- We can create a high-level API like expand_nodes() which will copy the vector from a source coordinate to all the coordinates in a bounding box.

I will keep thinking about what high-level API we need.  I am positive for moving the for-loops into API.

Reply:
Sounds good.  Let's land this and keep iterating on the higher-level api.